Fails to build with MSVC

Doing pip install -U pyresidfp yields the following:

```
  FAILED: CMakeFiles/_pyresidfp.dir/src/residfp/resample/SincResampler.cpp.obj
  "D:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29333\bin\Hostx86\x64\cl.exe"  /nologo /TP -DHAVE_CONFIG_H -DPROJECT_VERSION -D_pyresidfp_EXPORTS -IC:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\_skbuild\win-amd64-3.9\cmake-build -IC:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp -IC:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\src\residfp -IC:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\src\residfp\resample -I"d:\program files\python39\Include" -IC:\Users\Karol\AppData\Local\Temp\pip-build-env-dw84bquk\overlay\Lib\site-packages\pybind11\include /DWIN32 /D_WINDOWS /GR /EHsc /O2 /Ob2 /DNDEBUG -MD /bigobj /MP /GL /showIncludes /FoCMakeFiles\_pyresidfp.dir\src\residfp\resample\SincResampler.cpp.obj /FdCMakeFiles\_pyresidfp.dir\ /FS -c C:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\src\residfp\resample\SincResampler.cpp
  C:\Users\Karol\AppData\Local\Temp\pip-install-fpar1ybc\pyresidfp\_skbuild\win-amd64-3.9\cmake-build\siddefs-fp.h(34): fatal error C1017: invalid integer constant expression
```

--- a/.github/workflows/build_wheels.yml
+++ b/.github/workflows/build_wheels.yml
@@ -9,7 +9,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-20.04, macos-10.15]
+        os: [ubuntu-20.04, macos-10.15, windows-2019]
 
     steps:
       - uses: actions/checkout@v2
@@ -26,7 +26,7 @@ jobs:
         uses: pypa/cibuildwheel@v2.2.2
         env:
           CIBW_BUILD: "{cp,pp}3*-*"
-          CIBW_SKIP: "*-win32 *-win_amd64 *-manylinux_i686 *-musllinux_i686"
+          CIBW_SKIP: "*-win32 *-manylinux_i686 *-musllinux_i686"
           CIBW_ARCHS_MACOS: x86_64 universal2
           CIBW_ARCHS_LINUX: auto aarch64
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,10 +32,21 @@ include(CheckIncludeFileCXX)
 
 enable_testing()
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(USE_CLANG TRUE)
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  set(USE_GCC TRUE)
+endif()
+
+if(USE_GCC OR USE_CLANG)
+  check_include_file_cxx(mmintrin.h HAVE_MMINTRIN_H)
+else()
+  set(HAVE_MMINTRIN_H FALSE)
+endif()
+
 check_cxx_source_compiles("
 int main(void) { if (__builtin_expect(0, 0)) return 1; return 0; }
 " HAVE_BUILTIN_EXPECT)
-check_include_file_cxx(mmintrin.h HAVE_MMINTRIN_H)
 
 set(CMAKE_CXX_STANDARD 11)
 set(PACKAGE_VERSION ${PROJECT_VERSION})

--- a/src/pyresidfp.cpp
+++ b/src/pyresidfp.cpp
@@ -127,7 +127,9 @@ PYBIND11_MODULE(_pyresidfp, m) {
                with the audio mixer and output stage, including details on intended
                relative resistor values. Also included are schematics for the NMOS FET
                voltage controlled resistors (VCRs) used to control cutoff frequency, the
-               DAC which controls the VCRs, the NMOS op-amps, and the output buffer.
+               DAC which controls the VCRs, the NMOS op-amps, and the output buffer.)pbdoc"
+	       R"pbdoc(
+
 
 
                SID filter / mixer / output
@@ -205,7 +207,8 @@ PYBIND11_MODULE(_pyresidfp, m) {
                as the apparant resistance increases with increasing drain-to-source
                voltage. If the drain-to-source voltage should approach the gate voltage
                of 12V, the NMOS FET will enter saturation mode (a.k.a. active mode), and
-               the NMOS FET will not operate anywhere like a resistor.
+               the NMOS FET will not operate anywhere like a resistor.)pbdoc"
+	       R"pbdoc(
 
 
 
@@ -257,7 +260,8 @@ PYBIND11_MODULE(_pyresidfp, m) {
 
                Note that these are only approximate values for one particular SID chip,
                due to process variations the values can be substantially different in
-               other chips.
+               other chips.)pbdoc"
+	       R"pbdoc(
 
 
 
@@ -280,7 +284,8 @@ PYBIND11_MODULE(_pyresidfp, m) {
 
                Furthermore, the control of the two VCRs imposes a load on the DAC output
                which varies with the input signals to the VCRs. This can be seen from the
-               VCR figure above.
+               VCR figure above.)pbdoc"
+	       R"pbdoc(
 
 
 
@@ -329,7 +334,8 @@ PYBIND11_MODULE(_pyresidfp, m) {
 
                Provided a reasonably high input impedance and a reasonably low output
                impedance, the "op-amp" can be modeled as a voltage transfer function
-               mapping input voltage to output voltage.
+               mapping input voltage to output voltage.)pbdoc"
+	       R"pbdoc(
 
 
 
@@ -418,7 +424,8 @@ PYBIND11_MODULE(_pyresidfp, m) {
                                +---o--[A>--o------o           o--o--[A>--o-- vo (AUDIO OUT)
                                                D1 o--/ --4R2--o (4.25R2)
                                       $18         |           |
-                                      0=open   D0 +--/ --8R2--+ (8.75R2)
+                                      0=open   D0 +--/ --8R2--+ (8.75R2))pbdoc"
+	       R"pbdoc(
 
 
 
@@ -479,7 +486,8 @@ PYBIND11_MODULE(_pyresidfp, m) {
                These data indicate that the following function for 1/Q has been
                modeled in the MOS 8580:
 
-                  1/Q = 2^(1/2)*2^(-x/8) = 2^(1/2 - x/8) = 2^((4 - x)/8)
+                  1/Q = 2^(1/2)*2^(-x/8) = 2^(1/2 - x/8) = 2^((4 - x)/8))pbdoc"
+	       R"pbdoc(
 
 
 
@@ -533,7 +541,8 @@ PYBIND11_MODULE(_pyresidfp, m) {
                                |           R         |        GND
                                            |
                               GND                   GND
-                                          GND
+                                          GND)pbdoc"
+	       R"pbdoc(
 
 
 

--- a/src/residfp/siddefs-fp.h.in
+++ b/src/residfp/siddefs-fp.h.in
@@ -21,10 +21,10 @@
 #define SIDDEFS_FP_H
 
 // Compilation configuration.
-#define RESID_BRANCH_HINTS @RESID_BRANCH_HINTS@
+#cmakedefine01 RESID_BRANCH_HINTS
 
 // Compiler specifics.
-#define HAVE_BUILTIN_EXPECT @HAVE_BUILTIN_EXPECT@
+#cmakedefine01 HAVE_BUILTIN_EXPECT
 
 #ifndef M_PI
 #  define M_PI    3.14159265358979323846
@@ -56,7 +56,7 @@ const char* residfp_version_string = "@PACKAGE_VERSION@";
 }
 
 // Inlining on/off.
-#define RESID_INLINING @RESID_INLINING@
+#cmakedefine01 RESID_INLINING
 #define RESID_INLINE @RESID_INLINE@
 
 #endif // SIDDEFS_FP_H

